service/common/src/main/java/org/apache/polaris/service/catalog/iceberg/IcebergCatalog.java

```diff
@@ -232,7 +222,16 @@ public String name() {
   }
 
   @VisibleForTesting
-  public FileIO getIo() {
+  public synchronized FileIO getIo() {
+    if (catalogFileIO == null) {
+      LOGGER.debug(
+          "Initializing a default catalogFileIO with properties {}", tableDefaultProperties);
+      catalogFileIO = loadFileIO(ioImplClassName, tableDefaultProperties);
```

Comment:
This used to be performed only when `INITIALIZE_DEFAULT_CATALOG_FILEIO_FOR_TEST` was `true`. Now it appears to be performed in all cases... do I read it correctly? If so, what's the rationale for this behaviour change?

Reply:
`getIo()` and all of its callers are test-only.

Reply:
Noted. Please make it package-private.

Reply:
Yeah, I actually tried that. Turns out some tests will fail due to across package reference.

Reply:
But, why not have tests create FileIO instances and inject them into `IcebergCatalog`? I think it would be a lot more clear.

Reply:
What do you mean by "inject"? A setter will face exactly the same issue, which is public and only used by tests. 
Or are you talking about a subclass?

Reply:
Yes, looking at the test code, `IcebergCatalog` objects are created by tests for tests, so I believe since we started this cleanup effort, it makes sense for these tests to own FileIO creation.

Reply:
The class `IcebergCatalog` belongs to package `org.apache.polaris.service.catalog.iceberg`, while `IcebergCatalogTest` belongs to package `org.apache.polaris.service.quarkus.catalog`. Are you suggesting a setter or a subclass? I think a subclass will make the test code overly complicated. Introducing a setter like `setFileIO(fileIO)` will face the same issue, which is the method is public and used only by tests.  

Please let me know which way you prefer. cc @dimas-b @snazy.

Reply:
I'd suggest either a test-specific constructor or test-specific `initialize(...)` method, which would then accept a specific FileIO implementation.

Reply:
Added a sub class so that the only the test sub class creates load the fileIO